x-pack/plugins/security/public/account_management/user_profile/user_profile_api_client.ts

```diff
@@ -94,7 +94,8 @@ export class UserProfileAPIClient {
    */
   public bulkGet<D extends UserProfileData>(params: UserProfileBulkGetParams) {
     return this.http.post<Array<UserProfile<D>>>('/internal/security/user_profile/_bulk_get', {
-      body: JSON.stringify(params),
+      // Convert `Set` with UIDs to an array to make it serializable.
+      body: JSON.stringify({ ...params, uids: [...params.uids] }),
```

Comment:
__note:__ rationale behind using `Set` is covered in https://github.com/elastic/kibana/pull/138088#discussion_r937703419